genie-common/src/main/java/com/netflix/genie/common/dto/Cluster.java

```diff
@@ -37,6 +40,7 @@
 
     @NotNull(message = "A valid cluster status is required")
     private final ClusterStatus status;
+    private final Set<String> dependencies = new HashSet<>();
```

Comment:
As a general design comment for improving the code base I'm pretty sure now all the major dto's (JobRequest, Job, Cluster, Command, Application) now have dependencies as a field. As such this member variable should be pushed up into a super class that is shared so we don't have repetitive work. I thought this had already been done for some of them but dependencies should be pushed up as well.